tests/worker/test_workflow.py

```diff
@@ -7421,3 +7422,56 @@ async def test_workflow_dynamic_config_failure(client: Client):
         await assert_task_fail_eventually(
             handle, message_contains="Dynamic config failure"
         )
+
+
+@activity.defn
+async def raise_application_error(use_benign: bool):
+    if use_benign:
+        raise ApplicationError(
+            "This is a benign error", category=ApplicationErrorCategory.BENIGN
+        )
+    else:
+        raise ApplicationError(
+            "This is a regular error", category=ApplicationErrorCategory.UNSPECIFIED
+        )
+
+
+@workflow.defn
+class RaiseErrorWorkflow:
+    @workflow.run
+    async def run(self, use_benign: bool):
```

Comment:
```suggestion
    async def run(self, use_benign: bool) -> None:
```

Same on activity (though maybe `typing.NoReturn` there, didn't check). Should always set return types on activities/workflows (and most non-test methods)